sklearn/metrics/scorer.py

```diff
@@ -58,17 +121,46 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def __call__(self, estimator, X, y_true, sample_weight=None):
+        """Evaluate predicted target values for X relative to y_true.
+
+        Parameters
+        ----------
+        estimator : object
+            Trained estimator to use for scoring. Must have a predict_proba
+            method; the output of that is used to compute the score.
+
+        X : array-like or sparse matrix
+            Test data that will be fed to estimator.predict.
+
+        y_true : array-like
+            Gold standard target values for X.
+
+        sample_weight : array-like, optional (default=None)
+            Sample weights.
+
+        Returns
+        -------
+        score : float
+            Score function applied to prediction of estimator on X.
+        """
+        return self._score(partial(_method_caller, None), estimator, X, y_true,
```

Comment:
```suggestion
        return self._score(partial(_method_caller, cache=None), estimator, X, y_true,
```

Reply:
Since `cache` is a positional argument, `partial` needs to accept it as a positional argument.